static/app/views/performance/newTraceDetails/traceModels/traceTreeNode/transactionNode.tsx

```diff
@@ -0,0 +1,347 @@
+import type {Theme} from '@emotion/react';
+import * as Sentry from '@sentry/react';
+
+import type {Client} from 'sentry/api';
+import {pickBarColor} from 'sentry/components/performance/waterfall/utils';
+import {t} from 'sentry/locale';
+import type {EventTransaction} from 'sentry/types/event';
+import type {Organization} from 'sentry/types/organization';
+import {getStylingSliceName} from 'sentry/views/explore/tables/tracesTable/utils';
+import {TransactionNodeDetails} from 'sentry/views/performance/newTraceDetails/traceDrawer/details/transaction';
+import type {TraceTreeNodeDetailsProps} from 'sentry/views/performance/newTraceDetails/traceDrawer/tabs/traceTreeNodeDetails';
+import {isTransactionNode} from 'sentry/views/performance/newTraceDetails/traceGuards';
+import type {TraceTree} from 'sentry/views/performance/newTraceDetails/traceModels/traceTree';
+import type {TraceTreeNode} from 'sentry/views/performance/newTraceDetails/traceModels/traceTreeNode';
+import type {TraceRowProps} from 'sentry/views/performance/newTraceDetails/traceRow/traceRow';
+import {TraceTransactionRow} from 'sentry/views/performance/newTraceDetails/traceRow/traceTransactionRow';
+import type {TracePreferencesState} from 'sentry/views/performance/newTraceDetails/traceState/tracePreferences';
+
+import {BaseNode, type TraceTreeNodeExtra} from './baseNode';
+import {traceChronologicalSort} from './utils';
+
+const {info, fmt} = Sentry.logger;
+
+export class TransactionNode extends BaseNode<TraceTree.Transaction> {
+  private _spanPromises: Map<string, Promise<EventTransaction>> = new Map();
+  private _fromSpans: typeof TraceTree.FromSpans;
+  private _applyPreferences: typeof TraceTree.ApplyPreferences;
+
+  extra: TraceTreeNodeExtra;
+
+  searchPriority = 1;
+
+  constructor(
+    parent: BaseNode | null,
+    value: TraceTree.Transaction,
+    extra: TraceTreeNodeExtra,
+    fromSpans: typeof TraceTree.FromSpans,
```

Comment:
This won't be needed. fromSpans should be a tree method, instead of being static. It always requires updating the tree bounds. Also applyPreferences below is only can be called in  `fromSpans` in traceTree.tsx. We won't have to pass statics as params like this, this seems like an antipattern.